src/hooks/useAuth.tsx

```diff
@@ -40,12 +44,32 @@ export const useAuth = (): AuthContextType => {
 export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
   const [isAuthenticated, setIsAuthenticated] = useState<boolean | null>(null);
   const [error, setError] = useState<string>('');
+  const [member, setMember] = useState<Member | null>(null);
   const [loading, setLoading] = useState<boolean>(true);
 
   useEffect(() => {
     getSession();
   }, []);
 
+  useEffect(() => {
+    if (isAuthenticated) {
+      getCurrentUser()
+        .then((mem) => {
+          if (mem) {
+            setMember(mem);
+          } else {
```

Comment:
`getCurrentUser` returns `Member | null`. I can make it return `undefined` instead of `null` if you want, but I much prefer returning errors as values over throwing exceptions. If you have a good justification though I'm willing to change my mind

Reply:
If it returns member | null why do we have an if else branch? Can just `setMember(mem)` can't we?

Reply:
Ahhh now it makes sense